mem.go

```diff
@@ -80,6 +82,33 @@ func (r RO) EqualString(s string) bool { return r.str() == s }
 // the same bytes.
 func (r RO) EqualBytes(b []byte) bool { return r.str() == string(b) }
 
+var builderPool = sync.Pool{
+	New: func() interface{} {
+		return new(strings.Builder)
+	},
+}
+
+// StringCopy returns m's contents in a newly allocated string.
+func (r RO) StringCopy() string {
+	buf := builderPool.Get().(*strings.Builder)
+	buf.WriteString(r.str())
+	s := buf.String()
+	buf.Reset()
```

Comment:
likewise defer here and then just `return buf.String()` ?

Reply:
No defer saves ~6% of execution time, even using a newer compiler. And it's not clear to me that it improves readability that much--the function is six lines long, of which two are defers.

Reply:
But OK, done. I can't merge, so will leave button-pushing to you.